### ISSUE TITLE
Please declare a license for the codebase

### ISSUE THREAD

**Issue by felipesanches** [ppeccin/WebMSX#4]
Several files mention this in the header:
// Copyright 2015 by Paulo Augusto Peccin. See license.txt distributed with this file.

But I could not find a license.txt file anywhere.

**Comment by ppeccin** [comment]
Yes, I'm still working on that! I'll let you know when its ready.
Do you have any doubts or concerns regarding the license?
Please let me know if I can help you.

**Comment by felipesanches** [comment]
yes, I do have a doubt: What is the license you chose for this codebase?

**Comment by ppeccin** [comment]
As I said, I did not settle on a specific license yet. Still working in that, analyzing possibilities.

Meanwhile, if you have any doubts regarding what you can and what you cannot do with it, please just ask me.

**Comment by maly** [comment]
Hi ppeccin, I would to embed your MSX emulator in my online assembler as a test machine. Is it possible? (I prefer MIT license over GPL)

**Comment by ppeccin** [comment]
Yes, of course!
As long as you maintain the overall original experience and include credits the project.

**Comment by maly** [comment]
No, it is not specific to MSX, it is an online assembler for 8bit processors 8080, 6502, 6800, 6809, Z80 and 1802, with some possibilities of testing programs in emulators (ZX Spectrum, KIM-1 etc.) MSX should be very nice opportunity for further testing. Look at http://www.asm80.com

**Comment by ppeccin** [comment]
Nice! Also, have a look at msxpen.com

**Comment by felipesanches** [comment]
@ppeccin Everything could be much simpler if you simply added a LICENSE.txt file to the repository. Do you need help choosing the best license for this project ? Feel free to ask and I'll try to help.

**Comment by ppeccin** [comment]
Yes, please. Which licence you suggest? I want the project to be free, letting people use it freely, but maintaining the original experience, look and feels, and credits to the author and original project page.

**Comment by maly** [comment]
In my opinion and experience: You have three main options: BSD-like license (MIT), GPL-like (GNU GPL) or Creative Commons.

Credits to the author and original project page is mandatory for all licences (MIT, GPL, CC-BY).

GPL allows everybody to take, change and distribute as long as it remain under the GPL. It prohibits embedding to any non-GPL software bundle. 

BSD (MIT) allows everybody to take, change and distribute freely, if author's credits are preserved.

CC licenses has three main flavors: NC (Non Commercial - means nobody can sell it), ND (Non Derivative - means "you can use it, but do not change it") and SA (Share Alike - means "use it, change it, but let it open"). You can combine these attributes, e.g. CC-BY-ND-SA (BY = preserve credits, ND = make no changes, SA = your software should be open source too)

Just one license maintain original experience, look & feel: CC-BY-ND. But it prohibits other people create their own creations upon your code. They can embed it "as-it-is", but they can't make any changes.

The way I personally prefer is the MIT license. It is very free and open, saying "Do anything you want as long as you preserve my credits for original code". You could use MIT too with some appendix like "preserve my look & feel".

So, there are two ways... CC-BY-ND allows free using, but no changes to anything. MIT allows free using and allows other to modify your code. You can ask them to preserve original feeling.

**Comment by maly** [comment]
(In my case, I would to embed your emulator in my page with your design and feel, but I have to make some changes to your codebase, which allows my IDE to inject compiled code into MSX memory, read memory & processor state or perform single CPU step)

**Comment by ppeccin** [comment]
@maly So, to allow your usage (which I think its ok), it would be MIT, right?
(btw, the emulator has a JS API to allow most of the things you need with no changes, but is not documented yet)

What I don't want people doing is just get the project, change logos and interface details, and show it as if it were a different thing.

**Comment by maly** [comment]
Yes, MIT is perfect. JS API is great thing, I really appreciate it.

Unfortunatelly you can not prevent such acting in an open-source world. Anyone can take your OSS, slightly change and tells it is different thing. But the basic license rule is: "Tell everybody who is the original author", so people can see the original thing and make their opinion. 

If you want to hold some of your rights, especially "do not change my software", you have to use CC-BY-ND (Non-derivative license). In this case everyone can use your emulator freely, but they may not change anything.

**Comment by felipesanches** [comment]
But then, if you forbid commecial usage or if you forbid modifications to the program, then it is not considered free software / open source software.

**Comment by ppeccin** [comment]
E o que vcs acham da "GNU AFFERO GENERAL PUBLIC LICENSE"?
Foi a que eu acabei usando no Javatari, meu outro projeto de emulador. Mas nem sei se é a melhor escolha.

**Comment by felipesanches** [comment]
@ppeccin 's question was:

And what do you think of the "GNU AFFERO GENERAL PUBLIC LICENSE"?
This was the one I ended up choosing for Javatari, my other emulator project. But I'm not sure it's the best choice

**Comment by felipesanches** [comment]
> @ppeccin 's question was:
>
> And what do you think of the "GNU AFFERO GENERAL PUBLIC LICENSE"?
> This was the one I ended up choosing for Javatari, my other emulator project. But I'm not sure it's > the best choice

I like the Affero GPL a lot for web apps. I think it could be a good choice if you like it as well.

**Comment by maly** [comment]
Affero GPL means "you have to publish your changes you made to this SW, even just for run on your web"

**Comment by ppeccin** [comment]
Then I guess it can be too restrictive...

**Comment by felipesanches** [comment]
http://www.wtfpl.net/txt/copying/

```
            DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
                    Version 2, December 2004

 Copyright (C) 2004 Sam Hocevar <sam@hocevar.net>

 Everyone is permitted to copy and distribute verbatim or modified
 copies of this license document, and changing it is allowed as long
 as the name is changed.

            DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE
   TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION

  0. You just DO WHAT THE FUCK YOU WANT TO.
```

**Comment by felipesanches** [comment]
Alternatively you may prefer to use this other one:

"THE BEER-WARE LICENSE" (Revision 42): <phk@FreeBSD.ORG> wrote this file. As long as you retain this notice you can do whatever you want with this stuff. If we meet some day, and you think this stuff is worth it, you can buy me a beer in return Poul-Henning Kamp 

https://spdx.org/licenses/Beerware.html

**Comment by FiXato** [comment]
@maly, you wrote:

> In my opinion and experience: You have three main options: BSD-like license (MIT), GPL-like (GNU GPL) or Creative Commons.

However, Creative Commons actually [recommends _against_ using their licenses for software](https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software):

> We recommend against using Creative Commons licenses for software. Instead, we strongly encourage you to use one of the very good software licenses which are already available. We recommend considering licenses made available by the Free Software Foundation or listed as “open source” by the Open Source Initiative.

**Comment by FiXato** [comment]
@ppeccin you could have a look at https://choosealicense.com/ and see which license that would come up with for your desired limitations and freedoms.
